components/MobileNavigation.js

```diff
@@ -20,23 +20,53 @@ export default class MobileNavigation extends Component {
 
   state = {
     showMobileNav: false,
-    expandMobileNav: false
+    expandMobileNav: false,
+    nowSelectMenu: null,
+    scrollPosition: null
   }
 
   componentDidMount () {
+    const { navs } = this.props
     window.addEventListener('scroll', this.handleScroll)
+    const scrollPosition = navs.reduce((obj, nav, index) => {
+      const navScroll = {
+        name: nav.label,
+        scroll: document.getElementById(nav.label.toLowerCase()).getBoundingClientRect().top + window.scrollY
+      }
+      obj[index] = navScroll
+      return obj
+    }, [])
+    this.setState({
+      scrollPosition: scrollPosition
+    })
   }
 
   componentWillUnmount () {
     window.removeEventListener('scroll', this.handleScroll)
   }
 
   handleScroll = (e) => {
+    const _docHeight = (document.height !== undefined) ? document.height : document.body.offsetHeight
+    let result
```

Comment:
Thanks for the question. What I mean is that you can write code in a way that does not require changing a variable. For example, you can do:

```js
    const selectedMenu = this.findSelectedMenuFromScroll()
    if (this.state.nowSelectMenu !== selectedMenu) {
      this.setState({ nowSelectMenu: selectedMenu })
    }
```

Then you can create a separate method like this:

```js
  findSelectedMenuFromScroll () {
    ...
    if (...) {
      return ...
    } else {
      return ...
    }
  }
```

By factoring out functions, you can eliminate variables that can change.